### PR TITLE
locals as entity

### PR DESCRIPTION
**Pull request by Gnito**

### CODE REVIEW

File: app/controllers/transactions_controller.rb [hunk 1]
```diff
@@ -36,22 +36,41 @@ def show
 
     MarketplaceService::Transaction::Command.mark_as_seen_by_current(params[:id], @current_user.id)
 
-    show_total_price = tx[:payment_process] != :none
-    total_price_label = (tx[:payment_process] != :preauthorize) ? t("transactions.price") : t("transactions.total")
-    unit_type = listing[:unit_type].present? ? translate_quantity_unit(tx[:unit_type]) : nil
+    price_break_down_locals =
```
Comment (sktoiva) [inline app/controllers/transactions_controller.rb]: This if-else conditional should be in it's own method. In my opinion, it's difficult to read code where there is a 22 line conditional statement with return value stored in a variable. 

Refactoring this into a own private method would even render the need for this local variable unnecessary.

File: app/controllers/transactions_controller.rb [hunk 2]
```diff
@@ -36,22 +36,41 @@ def show
...
+      if tx[:payment_process] == :none && tx[:listing_price].cents == 0
+        nil
+      else
```
Comment (sktoiva) [inline app/controllers/transactions_controller.rb]: And unless you  want to be explicit:

```ruby
if tx[:payment_process] != :none || tx[:listing_price].cents != 0
```
would remove the else branch.

File: app/controllers/transactions_controller.rb (hunk 1 above)
Comment (Gnito) [inline app/controllers/transactions_controller.rb]: true dat

File: app/controllers/transactions_controller.rb (hunk 2 above)
Comment (Gnito) [inline app/controllers/transactions_controller.rb]: Yeah, I wanted to be explicit

**Pull request merged**